feat(website): add a warning admonition for unique identifiers (#8908)
elastic/eui: packages/website/docs/components/tables/basic.mdx

Reviewer: 💭 Imho, most of the text wording changes here were not needed. But that's subjective and I don't mind the changes considering nothing changes much.

Author: That's fair. To me, the flow is better, the wording is simpler. It's easier to read.

### Sorting

The following example shows how to set up column sorting using the `sorting` property and marking sortable columns with `sortable: true`. To enable default sorting for **every** column, pass `enableAllColumns: true` to the `sorting` prop. If you don't want users to control the sort, you can pass `readOnly: true` to the `sorting` prop or to individual columns.

:::warning Unique identifiers required for sorting

> When using sorting with **EuiBasicTable**, make sure each row has a unique identifier. Duplicate identifiers can cause sorting to display incorrect results, such as showing fewer items than expected. Ensure your data has unique `id` values.

Reviewer: Nit: Imho the last sentence feels a bit odd at the end and like a duplication. Maybe we can combine it with the first sentence to clarify.

```suggestion
When using sorting with **EuiBasicTable**, make sure each row has a unique identifier by passing unique `id` values on your data. Duplicate identifiers can cause sorting to display incorrect results, such as showing fewer items than expected.
```

Author: Good point. How about:

```suggestion
When using sorting with **EuiBasicTable**, make sure each row has a unique identifier by passing a unique `id` values. Duplicate identifiers can cause sorting to display incorrect results, such as showing fewer items than expected. Ensure your data has unique `id` values.
```

> When using sorting with **EuiBasicTable**, make sure each row has a unique identifier by passing unique `id` values. Duplicate identifiers can cause sorting to display incorrect results, such as showing fewer items than expected. Ensure your data has unique `id` values.

Author: ```suggestion
When using sorting with **EuiBasicTable**, make sure each row has a unique identifier by passing unique `id` values. Duplicate identifiers can cause sorting to display incorrect results, such as showing fewer items than expected.
```
